lib/client.js

```diff
@@ -489,6 +461,245 @@ class Client {
+  resourceSearch({ resourceType, searchParams } = {}) {
+    const relativePathWithQuery = `${resourceType}?${new URLSearchParams(searchParams)}`;
+    return this.httpClient.get(relativePathWithQuery);
+  }
+
+  /**
+   * Search across all FHIR resource types in the system.
+   * Only the parameters defined for all resources can be used.
+   *
+   * @example
+   *
+   * // Using promises
+   * fhirClient.systemSearch({
+   *   searchParams: { name: 'smith' }
+   * }).then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.systemSearch({ searchParams: { name: 'smith' } });
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {Object} params.searchParams - The search parameters.
+   *
+   * @return {Promise<Object>} FHIR resources in a FHIR Bundle structure.
+   */
+  systemSearch({ searchParams } = {}) {
+    return this.httpClient.get(`/_search?${new URLSearchParams(searchParams)}`);
+  }
+
+  /**
+   * Search for FHIR resources within a compartment.
+   * The resourceType and id must be specified.
+   *
+   * @example
+   *
+   * // Using promises
+   * fhirClient.compartmentSearch({
+   *   resourceType: 'Observation',
+   *   compartment: { resourceType: 'Patient', id: 123 },
+   *   searchParams: { code: 'abc' }
+   * }).then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.compartmentSearch({
+   *   resourceType: 'Observation',
+   *   compartment: { resourceType: 'Patient', id: 123 },
+   *   searchParams: { code: 'abc' }
+   * });
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {Object} params.searchParams - The search parameters.
+   *
+   * @return {Promise<Object>} FHIR resources in a FHIR Bundle structure.
+   */
+  compartmentSearch({ resourceType, compartment, searchParams } = {}) {
+    let compartmentType;
+    let compartmentId;
+
+    if (compartment) {
+      compartmentType = compartment.resourceType;
+      compartmentId = compartment.id;
+    }
+
+    const query = new URLSearchParams(searchParams);
+    const relativePath = `/${compartmentType}/${compartmentId}/${resourceType}?${query}`;
+
+    return this.httpClient.get(relativePath);
+  }
+
+  /**
+   * Retrieve the change history for a FHIR resource id, a resource type or the entire system
+   *
+   * @example
+   *
+   * // Using promises
+   * fhirClient.history({ resourceType: 'Patient', id: '12345' });
+   *   .then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.history({ resourceType: 'Patient', id: '12345' });
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {string} params.resourceType - The resource type (e.g. "Patient", "Observation").
+   * @param {string} params.id - The FHIR id for the resource.
+   *
+   * @return {String} level - The history level
```

Comment:
History and search wouldn't return bundles?

Reply:
I pushed new commits to move the logic for checking search and history levels in the test, and restored the return value to `Bundle` in the relevant examples